homeassistant/components/proxmoxve/__init__.py

```diff
@@ -26,114 +26,74 @@
-                                            cv.positive_int
-                                        ],
-                                        vol.Optional(CONF_CONTAINERS, default=[]): [
-                                            cv.positive_int
-                                        ],
-                                    }
-                                )
-                            ],
-                        ),
-                    }
-                )
-            ],
-        )
-    },
-    extra=vol.ALLOW_EXTRA,
-)
+CONFIG_SCHEMA = cv.deprecated(DOMAIN)
 
 
 async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up the platform."""
+
+    if DOMAIN not in config:
+        return True
+
     hass.data.setdefault(DOMAIN, {})
 
-    def build_client() -> ProxmoxAPI:
-        """Build the Proxmox client connection."""
-        hass.data[PROXMOX_CLIENTS] = {}
+    hass.data[PROXMOX_CLIENTS] = {}
 
-        for entry in config[DOMAIN]:
-            host = entry[CONF_HOST]
-            port = entry[CONF_PORT]
-            user = entry[CONF_USERNAME]
-            realm = entry[CONF_REALM]
-            password = entry[CONF_PASSWORD]
-            verify_ssl = entry[CONF_VERIFY_SSL]
+    for entry in config[DOMAIN]:
+        host = entry[CONF_HOST]
+        port = entry.get(CONF_PORT, DEFAULT_PORT)
+        user = entry[CONF_USERNAME]
+        realm = entry.get(CONF_REALM, DEFAULT_REALM)
+        password = entry[CONF_PASSWORD]
+        verify_ssl = entry.get(CONF_VERIFY_SSL, DEFAULT_VERIFY_SSL)
 
-            hass.data[PROXMOX_CLIENTS][host] = None
+        hass.data[PROXMOX_CLIENTS][host] = None
 
-            try:
-                # Construct an API client with the given data for the given host
-                proxmox_client = ProxmoxClient(
-                    host, port, user, realm, password, verify_ssl
-                )
-                proxmox_client.build_client()
-            except AuthenticationError:
-                _LOGGER.warning(
-                    "Invalid credentials for proxmox instance %s:%d", host, port
-                )
-                continue
-            except SSLError:
-                _LOGGER.error(
-                    "Unable to verify proxmox server SSL. "
-                    'Try using "verify_ssl: false" for proxmox instance %s:%d',
-                    host,
-                    port,
-                )
-                continue
-            except ConnectTimeout:
-                _LOGGER.warning("Connection to host %s timed out during setup", host)
-                continue
-            except requests.exceptions.ConnectionError:
-                _LOGGER.warning("Host %s is not reachable", host)
-                continue
+        try:
+            # Construct an API client with the given data for the given host
+            proxmox_client = ProxmoxClient(
+                host, port, user, realm, password, verify_ssl
+            )
```

Comment:
`async_setup` should not be used to set up the integration, instead, it should trigger an import step in the config flow.
After import, that `async_setup_entry` will handle the imported entry.